### DISCUSSION TITLE
What's the difference between docs and Wikis?

### DISCUSSION THREAD

**Discussion by Rod-at-DOH** [community/community#141193] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

(First of all, I'm sorry if I've chosen the wrong forum to ask this question. None of the others seemed appropriate, but I could be wrong.)

I am writing some documentation for the developers, Project Managers, etc., on how to use GitHub in a way consistent with how people currently use it. For example, I've discussed the Wikis section of a repository. The way I understand Wikis, it is where developers and maybe PMs will record different aspects of the software solution that is in the repo, decisions they made along the way, etc. So, other people not members of that repo can know the thinking of the software team as they worked on it.

However, I was recently reminded of the `docs` folder at the root of the repo. `Docs` is also used for documenting the system. Of course, you can put Markdown files within `docs`, but also Word, PowerPoint, PDFs, etc. Anything and everything that may pertain to the system.

This got me wondering, what's the differences between how the Wiki and the `docs` folder are to be used? There seems to be a significant overlap, but is there an something different that each is meant to be used?

**Comment by t1noo7** [comment, marked as answer]
Looks like a great question! You're not alone in trying to clarify the distinction between GitHub's Wiki feature and the docs folder within a repository. Both are intended to store documentation, but they serve different purposes and are typically used in different ways. I'll break down their roles and highlight the key differences:

1. GitHub Wiki
The Wiki feature in GitHub is designed as a collaborative space for documentation. It is accessible via the "Wiki" tab in the repository. Here are its key characteristics:

- Ease of Use: Wikis allow for quick and easy editing directly in the browser using a rich-text or Markdown editor. It's designed for rapid documentation updates without needing to modify the codebase.

- Separate from the Repo Code: The Wiki is separate from the main repository. This means it's not part of the version-controlled code base and doesn't require a developer to clone the repository or work with Git to edit the content. It's more of a "living document" that evolves as needed.

- Good for Ongoing Documentation: Wikis are often used for high-level, continuously updated documentation. Project Managers, developers, and other stakeholders can document the evolution of decisions, explain how features work, or describe processes. This documentation is generally expected to be read and updated by people throughout the project's lifecycle.

- Accessibility: The Wiki is more accessible to non-developers (e.g., PMs or business analysts) because it doesn’t require using Git to add or update content. It’s designed for collaboration among a broader audience.

- Limitations: Wikis are not tightly coupled with the version control of the repo. If the code evolves but the Wiki is not kept in sync, there can be discrepancies between the code and the documentation.

2. Docs Folder
The docs folder is part of the repository's main structure and is typically used for documentation that is version-controlled along with the codebase. Here’s how it differs:

- Version Control: Since the docs folder is part of the repository, every change to the documentation is tracked using Git, just like the code. This is ideal for documentation that needs to stay in sync with different versions of the software. For example, if your repository has different branches (say for different releases), the docs folder can reflect the documentation for each version or branch.

- Coupled with the Codebase: Unlike the Wiki, which is separate, the docs folder sits right inside the repo and is versioned along with the code. This is helpful when documentation must closely align with specific releases or features in the code.

- More Flexible Format: The docs folder can contain not just Markdown files, but any other formats (Word, PDF, PowerPoint, etc.) that may be required. This flexibility can be useful for different types of documentation that are not easily captured in a Wiki (e.g., architectural diagrams in PDF format, project plans in PowerPoint).

- Good for Static or Version-Specific Documentation: The docs folder is ideal for things like API documentation, system architecture descriptions, or release notes that are tightly bound to a specific version of the code.

- Maintenance: Editing files in the docs folder requires someone to work with Git (cloning the repo, making changes, committing, and pushing). This means that it’s less convenient for non-technical stakeholders compared to the Wiki.

-> Use Case Differences
Wiki: Use the Wiki for high-level, frequently updated documentation. This could include project overviews, guidelines for contributing, meeting notes, decision logs, feature overviews, or general discussions on the evolution of the software. It’s a dynamic resource meant for ongoing collaboration.

Docs Folder: Use the docs folder for technical, version-controlled documentation that must remain in sync with the codebase. Examples include API references, configuration files, release notes, or in-depth technical guides. The docs folder ensures that documentation evolves alongside the code.

(*) Common Practices
Wiki for General or Non-Versioned Documentation: Many teams use the Wiki for general project documentation, roadmaps, or planning notes that aren’t tightly coupled to specific versions of the code.

Docs Folder for Versioned Documentation: Detailed technical documentation like installation instructions, API specs, architecture diagrams, or system design documents often go into the docs folder. Teams may use this folder for documentation that must evolve in tandem with specific branches or versions of the software.

Linking Between the Two: It’s common for the Wiki and the docs folder to coexist. Some teams use the Wiki to offer general information, then link to the docs folder for technical details that are specific to the current release or version. For example, the Wiki might include a high-level overview of an API, while the docs folder contains the full, version-specific API documentation.


Overall, The key difference lies in version control and accessibility. Use the Wiki for easily editable, high-level, and collaborative documentation that isn't tightly bound to specific versions of the code. Use the docs folder for technical, versioned documentation that needs to evolve alongside the codebase.

Hopefully, this helps you clarify the distinction! :)

**Reply by Rod-at-DOH** [reply]
Outstanding, @t1noo7 !! I will use what you've written here in my documentation. I'll cite this discussion. Thank you!!

**Reply by t1noo7** [reply]
> Outstanding, @t1noo7 !! I will use what you've written here in my documentation. I'll cite this discussion. Thank you!!

You're welcome @Rod-at-DOH  , happy writing and have a nice day!!! 👍👍👍

**Reply by zenoli** [reply]
@Rod-at-DOH pro tip: Next time ask ChatGPT directly instead of waiting for a human to ask ChatGPT for you. Wayyyyy faster, trust me :-)
